### PR TITLE
Update docs to refer to Package Storage V2

### PR DESCRIPTION
**Pull request by mrodm**
## What does this PR do?

Update docs to remove references to Package Storage V1, as well as references to staging or snapshot environments.

To be merged after the switch to Package Storage V2 is done

### CODE REVIEW

**Review by jsoriano — APPROVED** [review]

File: docs/developer_workflow_bug_fix_older_stack_version.md [hunk 1]
```diff
@@ -30,17 +25,8 @@ For example: `backport-aws-7.15.0`.
 ![alt text](./images/create_branch.png)
 
 ### Step3: Create a PR for the bug fix
-Apply bugfixes and open a PR against the `<package_name>-<stack_version>` branch.
+Apply bugfixes and open a PR against the `backport-<package_name>-<stack_version>` branch.
 For example: I created this [PR](https://github.com/elastic/integrations/pull/4162) to fix
 the EC2 dashboard for Kibana 7.x.
```
Comment (mrodm) [inline docs/developer_workflow_bug_fix_older_stack_version.md:30]: @elastic/ecosystem if we set `stack_version` as part of the branch name. two differnt versions that need to be backported could be using the same stack version.

Not sure what other suggestion to be done here, maybe the major and minor version of the package

Comment (jsoriano) [reply docs/developer_workflow_bug_fix_older_stack_version.md:30]: Ah yes, I think that the version there should be the major and minor of the package, not the target stack.

Comment (mrodm) [reply docs/developer_workflow_bug_fix_older_stack_version.md:30]: In the case of this example it could be `backport-aws-1.19`

**Review by jsoriano — APPROVED** [review]

**Review by jsoriano — APPROVED** [review]

**Pull request merged**